testscripts/v4l2test.py

```diff
@@ -0,0 +1,107 @@
+def traverse(mediafile, decode, encode, mlist, aformat, handler):
+    if os.path.isfile(mediafile):
+        v4l2parse(mediafile, decode, encode, mlist, aformat, handler)
+    for root,dirs,files in os.walk(mediafile):
+        for testfile in files:
+            testfile = root + os.sep + testfile
+            if os.path.isfile(testfile):
+                v4l2parse(testfile, decode, encode, mlist, aformat, handler)
+def v4l2test(yamidirectory, yamidecode, yamiencode, v4l2decode, v4l2encode, capidecode, capiencode, mediafile, handler):
+    yamilist = ['-1', '0', '1', '2', '3', '4']
+    v4l2list = ['0', '3', '4']
+    capilist = ['-1', '0', '1', '2', '3', '4']
+    if v4l2config(yamidirectory, 'LIBYAMI_PREFIX', handler, 'tests'):
+        traverse(mediafile, yamidecode, yamiencode, yamilist, 'w', handler)
+        if v4l2config(yamidirectory, 'LIBYAMI_PREFIX', handler, 'tests', 'tests-gles'):
+            traverse(mediafile, yamidecode, yamiencode, yamilist, 'm', handler)
+    if v4l2config(yamidirectory, 'LIBYAMI_PREFIX', handler, 'tests', 'capi'):
+        traverse(mediafile, capidecode, capiencode, capilist, 'w', handler)
+        if v4l2config(yamidirectory, 'LIBYAMI_PREFIX', handler, 'tests', 'capi', 'tests-gles'):
+            traverse(mediafile, capidecode, capiencode, capilist, 'm', handler)
+    if v4l2config(yamidirectory, 'LIBYAMI_PREFIX', handler, 'tests', 'v4l2', 'tests-gles'):
+        traverse(mediafile, v4l2decode, v4l2encode, v4l2list, 'w', handler)
+        handler.write('\n')
+        traverse(mediafile, v4l2decode, v4l2encode, v4l2list, 'm', handler)
+if __name__=='__main__':
+    if len(sys.argv) < 2:
+        print('Pls input: python v4l2test.py document|dirctory')
+        sys.exit(-1)
+    if not os.path.exists(sys.argv[1]):
+        print(sys.argv[1] + ' does not exist')
+        sys.exit(-1)
+    mediafile = os.path.abspath(sys.argv[1])
+    decodepath = os.path.dirname(os.path.abspath(sys.argv[0]))
+    logdirectory = 'log'
+    logdirectory = os.path.join(decodepath, logdirectory)
+    if not os.path.isdir(logdirectory):
+        os.makedirs(logdirectory)
+    logfile = 'test-result-'+time.strftime('%Y-%m-%d-%H-%M-%S')+'.log'
+    logfile = logdirectory + os.sep + logfile
+    logfilehandler = open(logfile, 'a')
+    logfilehandler.write('v4l2 test result:\n')
+
+    yamidirectory = decodepath[0:decodepath.rfind('/')]
+    yamitestdirectory = decodepath[0:decodepath.rfind('/')]+os.sep+'tests'
+    yamidecode = yamitestdirectory + os.sep + 'yamidecode'
+    yamiencode = yamitestdirectory + os.sep + 'yamiencode'
+    v4l2decode = yamitestdirectory + os.sep + 'v4l2decode'
+    v4l2encode = yamitestdirectory + os.sep + 'v4l2encode'
+    capidecode = yamitestdirectory + os.sep + 'decodecapi'
+    capiencode = yamitestdirectory + os.sep + 'encodecapi'
+    v4l2test(yamidirectory, yamidecode, yamiencode, v4l2decode, v4l2encode, capidecode, capiencode, mediafile, logfilehandler)
```

Comment:
think about this 
v4l2test(xxx, "yamidecode", "yamiencode", modes, xxx)
v4l2test(xxx, "v4l2decode", "v4l2decode", modes, xxx)